linera-execution/src/execution_state_actor.rs

```diff
@@ -358,6 +358,11 @@ where
                 }
             }
 
+            ApplicationIndex { callback } => {
+                let index = self.txn_tracker.application_index();
```

Comment:
Maybe we should call these `NextApplicationIndex` and `next_application_index`. Otherwise the reader will assume that they refer to the current application itself, not to the next application it _creates_.

Reply:
Ok, done.